shell_substitution.h

```diff
@@ -0,0 +1,92 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#define COMM_SIZE 6
+
+// Display result of the assignment
+struct evaluation_factors {
+    bool valid;
+    char *assignment;
+};
+
+bool is_print_command_valid(char *line)
+{
+    // NOTE: Need to handle single quote as invalid
+    char *command = strndup(line, COMM_SIZE);
+
+    if (!strcmp(command, "print ")) return true;
+    return false;
+}
+
+char *retrieve_string_inside_quotation_mark(char *line)
+{
+    // Retrieve the string inside the quotation marks
+    char *display_str = (char *)malloc (sizeof (char) * strlen(line));
+    for (int i = (COMM_SIZE + 1); i < (strlen(line)-1); i++) 
+        display_str[i-(COMM_SIZE + 1)] = line[i];
+    return display_str;
+}
+
+bool is_dollar_present(char *str)
+{
+    for (int i = 0; i < strlen(str); i++)
+        if (str[i] == '$') return true;
+    return false;
+}
+
+int get_index_of_char_in_string(char *assignment)
+{
+    // Retrieve index of '=' sign
+    int index = 0;
+    for (int i = 0; i < strlen(assignment); i++) {
+        if (assignment[i] == '=') return i;
+    }
+    return index;
+}
+
+char *extract_substring_before_equal_sign(char *assignment)
+{
+    char *tmp_str = (char *)malloc (sizeof (char) * strlen(assignment));
+    for (int i = 0; i < strlen(assignment); i++) {
+        if (assignment[i] == '=') break;
+        tmp_str[i] = assignment[i];
+    }
+    return tmp_str;
+}
+
+char *extract_value_after_equal_sign(char *assignment, int start_idx)
```

Comment:
I believe this function does the same as extract_suffix_from_string.

Reply:
I understand what you mean. The reason I don't want to put everything in one right now is because I am afraid I might lose track of what is going on.

Reply:
I think multiple similar functions brings more uncertainty than general functions.